src/firewall/nft.rs

```diff
@@ -0,0 +1,1114 @@
+                            left: expr::Expression::Named(expr::NamedExpression::CT(expr::CT {
+                                key: "state".to_string(),
+                                family: None,
+                                dir: None,
+                            })),
+                            right: expr::Expression::List(vec![
+                                expr::Expression::String("established".to_string()),
+                                expr::Expression::String("related".to_string()),
+                            ]),
+                            op: stmt::Operator::IN,
+                        }),
+                        stmt::Statement::Accept(None),
+                    ],
+                ));
+                // Forward chain: ip saddr <subnet> accept
+                batch.add(make_rule(
+                    FORWARDCHAIN,
+                    vec![
+                        get_subnet_match(&subnet, "saddr", stmt::Operator::EQ),
+                        stmt::Statement::Accept(None),
+                    ],
+                ));
+                // Postrouting chain: ip saddr <subnet> jump <chain>
+                batch.add(make_rule(
+                    POSTROUTINGCHAIN,
+                    vec![
+                        get_subnet_match(&subnet, "saddr", stmt::Operator::EQ),
+                        get_jump_action(&chain),
+                    ],
+                ));
+            }
+        }
+
+        let rules = batch.to_nftables();
+
+        helper::apply_ruleset(&rules, None, None)?;
+
+        Ok(())
+    }
+
+    fn teardown_network(&self, tear: internal_types::TearDownNetwork) -> NetavarkResult<()> {
+        if !tear.complete_teardown {
+            log::info!("Nothing to tear down, network still in use");
+
+            return Ok(());
+        }
+
+        let mut batch = Batch::new();
+
+        let existing_rules = helper::get_current_ruleset(None, None)?;
+
+        if let Some(nets) = tear.config.subnets {
+            for subnet in nets {
+                // Match subnet, either saddr or daddr.
+                let match_subnet = |r: &schema::Rule| -> bool {
+                    // Statement matching: We only care about match statements.
+                    // Don't bother with left side. Just check if what they compare to is our subnet.
+                    for statement in &r.expr {
+                        match statement {
+                            stmt::Statement::Match(m) => match &m.right {
+                                expr::Expression::Named(expr::NamedExpression::Prefix(p)) => {
+                                    match *p.addr.clone() {
+                                        expr::Expression::String(s) => {
+                                            if *s == subnet.addr().to_string() {
+                                                return true;
+                                            }
+                                        }
```

Comment:
The wording says it checks the subnet but it actually only matches the network address (subnet.addr().to_string()) and thus strips off the the netmask, is this intentional?

Reply:
It's mostly me being lazy and figuring that there will never be two networks with the same network address but different subnet masks (given we have guarantees around that elsewhere), so we can just check half of the match payload.

Reply:
looks like p.len contains the subnet size
so something like `&& Into::<u32>::into(subnet.prefix_len()) == p.len` should match as well juts to be sure

Reply:
Done